wordpress-mobile/WordPress-iOS
# Implements progress bar for uploads in the pages list.

This PR implements the progress bar in the pages list.

There's no issue in the project for this, but it's a necessary step for full offline support.

## Important notes:

This PR implements new convenience classes `PostUploadProgressViewController` and `PostUploadProgressViewModel`, which will be used (in an upcoming PR) for posts too.

The reason why this PR doesn't make posts use these new classes is to avoid losing focus and making it harder to review the changes.

## Testing:

Testing is easier if you have Apple's Link Conditioner installed with a "Bad Network" configuration.

1. Go to the list of pages, and select the "Draft" filter (otherwise you own't see the post while it's uploading).
2. Create a new page, add an image to it (this is important for testing).
3. Publish it.

Notice that there's a progress bar for the page upload,  and that it moves ahead as it's uploading.

## PR submission checklist:
